troi/playlist.py

```diff
@@ -1,79 +1,148 @@
 import ujson
 import openpost
+import requests
 
-from troi import Recording
+from troi import Recording, Playlist, PipelineError, Element
 from troi.operations import is_homogeneous
-import troi
 
+LISTENBRAINZ_PLAYLIST_CREATE_URL = "https://test.listenbrainz.org/1/playlist/create"
 
-class PlaylistElement(troi.Element):
+def _serialize_to_jspf(playlist, created_for=None):
+
+    data = { "creator": "ListenBrainz Troi" }
+
+    if playlist.name:
+        data["title"] = playlist.name
+
+    if created_for:
+        data["created_for"] = created_for
+
+    tracks = []
+    for e in playlist.recordings:
+        track = {}
+        artist_mbids = [ str(mbid) for mbid in e.artist.mbids or [] ]
+        track["creator"] = e.artist.name if e.artist else ""
+        track["album"] = e.release.name if e.release else ""
+        track["title"] = e.name
+        track["identifier"] = "https://musicbrainz.org/recording/" + str(e.mbid)
+        if artist_mbids:
+            track["extension"] = {
+                "https://musicbrainz.org#jspf": {
+                    "artist_mbids" : artist_mbids,
+                }
+            }
+        tracks.append(track)
+
+    data['track'] = tracks
+
+    return { "playlist" : data }
+
+class PlaylistElement(Element):
+    """
+        Take a list of Recordings or Playlists and save, print or submit them.
```

Comment:
would be good to have some documentation that says that this actually stores a _list_ of playlists, and documentation in methods like `save` which say that if there are more than one that they're enumerated.

Add documentation saying that a list of Recordings is treated as a single playlist?

Reply:
done